airmozilla/starred/views.py

```diff
@@ -0,0 +1,102 @@
+import collections
+
+from django.shortcuts import render
+from django.views.decorators import cache
+from session_csrf import anonymous_csrf
+
+from jsonview.decorators import json_view
+from funfactory.urlresolvers import reverse
+
+from airmozilla.base.utils import paginate
+from airmozilla.starred.models import StarredEvent
+from airmozilla.main.models import (
+    Event,
+    CuratedGroup,
+)
+
+
+@cache.cache_control(private=True)
+@anonymous_csrf
+@json_view
+def sync_starred_events(request):
+    context = {'csrf_token': request.csrf_token}
+    if request.user.is_anonymous():
+        context['ids'] = []
+        return context
+    elif request.method == 'POST':
+        ids = request.POST.getlist('ids')
+        StarredEvent.objects.filter(user=request.user).exclude(
+            id__in=ids).delete()
+        for id in ids:
+            try:
+                event = Event.objects.get(id=id)
+                StarredEvent.objects.get_or_create(
+                    user=request.user,
+                    event=event
+                )
+            except Event.DoesNotExist:
+                # ignore events that don't exist but fail on other errors
+                pass
+
+    starred = StarredEvent.objects.filter(user=request.user)
+    context['ids'] = list(starred.values_list('event_id', flat=True))
+    return context
+
+
+def home(request, page=1):
+    template_name = 'starred/home.html'
+    ids = request.GET.getlist('ids')
+
+    if ids:
+        # how do we order_by the order in ids?
+        events = Event.objects.filter(id__in=ids).order_by('-created')
+        template_name = 'starred/events.html'
+
+    elif request.user.is_authenticated():
+        events = (
+            Event.objects.filter(starredevent__user=request.user.id)
+            .order_by('-starredevent__created')
+        )
```

Comment:
The sort order is wrong. It should have the most recently starred appear first. I.e. `'starredevent__created'`. I think.